File: cpp/LeetCode001_TwoSum.cpp
/*******************************************
 * FileNmae: cpp/LeetCode001_TwoSum.cpp
 * Author: Charles
 * Data: 11/4/2019, 1:29:43 AM
 * Data: f11a43aring
 ******************************************/
#include "leetcode_header.h"


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> testMap;
        for (int i = 0; i < nums.size(); i++) {
            int find_dest = target - nums[i];
            if (testMap.find(find_dest) == testMap.end()) {
                testMap[nums[i]] = i;
            }else {
                return vector<int> {testMap[find_dest], i};
            }
        }
        return vector<int> {0};
    }
};

int main()
{
    vector<int> testNums = { 2, 7, 11, 15 };
    int target = 9;
    Solution so;
    vector<int> result = so.twoSum(testNums, target);
    for (int i : result) {
        cout << i << endl;
    }
    return 0;
}

File: python3/LeetCode001_TwoSum.py
#!/bin/env python3
# LeetCode001_TwoSum.py
# @author zhaopeng
# @description 
# @created 2020-05-31T02:56:52.477Z+08:00
# @last-modified 2020-06-12T01:09:13.035Z+08:00
#


class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        tmp = {}
        for i in nums:
            if i in tmp.keys():
                return [tmp[i], i]
            else:
                tmp[target - i] = i
        return []

if __name__ == "__main__":
    so = Solution()
    nums = [2, 7, 11, 15]
    target = 9
    res = so.twoSum(nums, target)
    print(res)


        